DataDog/datadog-agent
# Adds installation information when using deb and rpm packages

### What does this PR do?

- Creates `install_info` file (if not already present) when installing the agent using`.deb` and `.rpm` packages, adding information about the installer version.
- Ensures the file is deleted after uninstall but it is kept untouched after upgrades.

### Motivation

- Allow users to track their installation methods
- Allows Datadog to make stats on installation methods

### Additional information

The `install_info` format should be:

```yaml
---
install_method:
  tool: [dpkg/rpm]
  tool_version: [dpkg/rpm]-<dpkg/rpm version number>
  installer_version: [deb/rpm/rpm_suse]_package(-iot)
```

In case `dpkg` or `rpm` are not available `tool` and `tool_version` would be replaced by `unknown`.

The rpm install_info is created in `posttrans` to avoid creating the file until the last moment, since rpm does not seem to recover from failed installs and we wouldn't want to leave the file laying around in that case.

### Describe your test plan

On each platform:
- install the package,
- check that the file was created correctly,
- remove the package,
- check that the file was removed,
- create a dummy install_info file,
- install the package again and
- check that the file was not modified this time.

Tested platforms:

- [x] Debian-based
-  [x] Debian-based (IoT)
- [x] RPM-based non-SUSE
- [x] RPM-based SUSE
- [x] RPM-based (IoT)
